mypy/api.py

```diff
@@ -60,3 +62,14 @@ def run(args: List[str]) -> Tuple[str, str, int]:
         sys.stderr = old_stderr
 
     return new_stdout.getvalue(), new_stderr.getvalue(), exit_status
+
+
+def run(args: List[str]) -> Tuple[str, str, int]:
+    # Lazy import to avoid needing to import all of mypy to call run_dmypy
+    from mypy.main import main
+    return _run(lambda: main(None, args=args))
+
+
+def run_dmypy(args: List[str]) -> Tuple[str, str, int]:
+    from mypy.dmypy import main
+    return _run(lambda: main(args))
```

Comment:
IIUC this doesn't support incremental flushing of errors (maybe we will do this at some point), but it is the same for normal runs, so probably OK. Maybe add a comment about this.